samza-test/src/main/java/org/apache/samza/test/framework/stream/CollectionStream.java

```diff
@@ -0,0 +1,189 @@
+ *<p>
+ * When sending messages using {@code CollectionStream<KV<K, V>>}, messages use K as key and V as message
+ * When sending messages using {@code CollectionStream<T>}, messages use a nullkey.
+ *</p>
+ * @param <T>
+ *        can represent a message with null key or a KV {@link org.apache.samza.operators.KV}, key of which represents key of a
+ *        {@link org.apache.samza.system.IncomingMessageEnvelope} or {@link org.apache.samza.system.OutgoingMessageEnvelope}
+ *        and value represents the message of the same
+ */
+public class CollectionStream<T> {
+
+  private String streamName;
+  private String systemName;
+  private CollectionStreamSystem system;
+  private Map<Integer, Iterable<T>> partitions;
+  private Map<String, String> streamConfig;
+  private static final String STREAM_TO_SYSTEM = "streams.%s.samza.system";
+
+  /**
+   * Constructs a new CollectionStream with multiple empty partitions from specified components.
+   * @param systemName represents name of the system stream is associated with
+   * @param streamName represents name of the stream
+   * @param partitionCount represents number of partitions, each of these partitions will be empty
+   */
+  private CollectionStream(String systemName, String streamName, Integer partitionCount) {
+    this.streamName = streamName;
+    this.streamConfig = new HashMap<>();
+    this.partitions = new HashMap<>();
+    this.systemName = systemName;
+    streamConfig.put(String.format(STREAM_TO_SYSTEM, this.streamName), systemName);
+    for (int i = 0; i < partitionCount; i++) {
+      partitions.put(i, new ArrayList<>());
+    }
+  }
+
+  /**
+   * Constructs a new CollectionStream with single partition from specified components.
+   * @param systemName represents name of the system stream is associated with
+   * @param streamName represents name of the stream
+   * @param partition represents the messages that the stream will be intialized with, default partitionId for the
+   *                  this single partition stream is 0
+   */
+  private CollectionStream(String systemName, String streamName, Iterable<T> partition) {
+    this.streamName = streamName;
+    this.streamConfig = new HashMap<>();
+    partitions = new HashMap<>();
+    this.systemName = systemName;
+    streamConfig.put(String.format(STREAM_TO_SYSTEM, this.streamName), systemName);
+    partitions.put(0, partition);
+  }
+
+  /**
+   * Constructs a new CollectionStream with multiple partitions from specified components.
+   * @param systemName represents name of the system stream is associated with
+   * @param streamName represents name of the stream
+   * @param partitions represents the partition state, key of the map represents partitionId and value represents
+   *                   the messages that partition will be initialized with
+   */
+  private CollectionStream(String systemName, String streamName, Map<Integer, ? extends Iterable<T>> partitions) {
```

Comment:
this makes the distinction clear, right? or am I missing something here??

Reply:
I'm referring to the "partitions" argument. The usage of "this" makes it correct, but since Java allows you to refer to instance variables both with "this" and without "this", it can get a bit harder to read if you then add an argument that is named the same as an instance variable.